### PR TITLE
Added allow_loading_unsigned_plugins management

### PR DESCRIPTION
**Pull request by themaroqa**
# Description

This PR is for adding the support for the `allow_loading_unsigned_plugins` feature

## Issues Resolved

https://github.com/sous-chefs/grafana/issues/395

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ This file is used to list changes made in each version of grafana.
 
 - Fixed docuemntation showing the incorrect action
 - Fixed a bug where ldap and grafana had hard coded paths.
+- Added `allow_loading_unsigned_plugins` management
```
Comment (xorima) [inline CHANGELOG.md]: Can you move this up to line 6 please 👍

File: documentation/grafana_config_plugins.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[back to resource list](https://github.com/sous-chefs/grafana#resources)
+
+---
+
+# grafana_config_plugins
+
+Configures the core plugins section of the configuration <https://grafana.com/docs/grafana/latest/administration/configuration/#plugins-1>
+
+Introduced: v9.5.2
```
Comment (xorima) [inline documentation/grafana_config_plugins.md]:
```suggestion
Introduced: v9.6.0
```

File: resources/config_plugins.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures the [plugins] section of the grafana config file
+
+property  :instance_name, String, name_property: true
+property  :allow_loading_unsigned_plugins, String, default: ''
```
Comment (xorima) [inline resources/config_plugins.rb]: I feel like this should be an array that we then join into a csv in code.

**Review by xorima — CHANGES_REQUESTED** [review]
Hey,

Thanks for this, a couple bits of feedback, mainly around the property should be an array which we then in code join into a csv

Thanks

**Review by ramereth — CHANGES_REQUESTED** [review]
Please add this suite to the GH Actions via `.github/workflows/ci.yml`.

File: resources/config_plugins.rb (hunk 3 above)
Comment (axelrtgs) [reply resources/config_plugins.rb]: Yeah i agree, we should take an array of names and join them in code. We should also update the example and tests to validate that the array was joined correctly

**Review by axelrtgs — COMMENTED** [review]
A recipe was added to run the code during the tests but no inspec was added to validate it did the right thing. should we have some validation tests or is successful converge ok here?

**Comment by themaroqa** [comment]
I should have fixed all the requests. Also added the inspec but for some reason I am missing it is failing even if testing locally it works.
If you note I am doing something wrong please tell me since I am definetely not tha main expert here

File: resources/config_plugins.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#
+# Cookbook:: grafana
+# Resource:: config_plugins
+#
+# Copyright:: 2018, Sous Chefs
```
Comment (ramereth) [inline resources/config_plugins.rb]:
```suggestion
# Copyright:: 2021, Sous Chefs
```

**Review by ramereth — CHANGES_REQUESTED** [review]
@themaroqa regarding the ChefSpec errors. It looks like you need to add `:grafana_config_paths` and `:grafana_config_plugins` to the `step_into` to get them to pass properly.

**Comment by themaroqa** [comment]
@ramereth what am I missing now?
`No instances for regex 'config_plugins-ubuntu-2004', try running 'kitchen list'
`
got this error for every platform

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -55,6 +55,7 @@ jobs:
           - 'basic'
           - 'configure'
           - 'azuread'
+          - 'config_plugins'
```
Comment (ramereth) [inline .github/workflows/ci.yml]: Test-kitchen translates `_` to `-` which is why this is failing.
```suggestion
          - 'config-plugins'
```

**Review by ramereth — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [9.6.0](https://github.com/sous-chefs/grafana/releases/tag/9.6.0)

**Pull request merged**